app/models/cart_validations.rb

```diff
@@ -46,6 +46,30 @@ def check_banned
     errors
   end
 
+  def check_consecutive # rubocop:disable AbcSize
+    errors = []
+    reserver = User.find_by(id: reserver_id)
+    models = get_items.keys
+    models.each do |model|
+      next unless model.max_per_user == 1 && model.max_checkout_length
```

Comment:
I think this has to be `maximum_checkout_length` (as well as below on line 59), which inherits from the category. Otherwise, this only takes into account model-specific length limits, not inherited limits.